sycl/include/sycl/builtins.hpp

```diff
@@ -521,7 +521,14 @@ detail::enable_if_t<
     detail::is_svgenfloat<T>::value && detail::is_genfloatptr<T2>::value, T>
 fract(T x, T2 iptr) __NOEXC {
   detail::check_vector_size<T, T2>();
-  return __sycl_std::__invoke_fract<T>(x, iptr);
+  // workaround to support multi_ptr without decoration
+  if constexpr (!iptr.is_decorated) {
```

Comment:
How would it work if `iptr` is something like `__global float *`? I'd expect we need to 
```
if constexpr(is_multi_ptr) 
  if constexpr (iptr.is_decorated)
```